modules/jupyter/jupyter_image/notebook_image/Dockerfile

```diff
@@ -1,3 +1,11 @@
 FROM jupyter/tensorflow-notebook:python-3.10
+USER root
 COPY requirements.txt ./requirements.txt
+RUN apt-get update && apt-get install -y --no-install-recommends openssh-server
+# Create a virtual environment
+RUN python3 -m venv /opt/venv
```

Comment:
What benefit does creating a virtual env do here?

Reply:
This approach addresses the following pip warning which was triggered due to `USER root`: 

`Step` #1: WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv`